### ISSUE TITLE
HTTP 429 issue, is there a way to fix it?

### ISSUE THREAD

**Issue by Brojustuserandomgenerationusername** [ytdl-org/youtube-dl#29773]
I did search it, im getting the issue out of nowhere and from what little research it was potentially due to Youtube graylisting IPs from accessing the videos, in my case it happens with any video i want to download and its not a thing about spamming video downloads as im every 4 hours trying with a single video and it keeps happening.

Short of switching IPs, is there any way to fix the issue? Im very simple when it comes to network handling and im not sure i know how to change my IP in the router. Using windows 10 if it means anything.

**Comment by 89z** [comment]
If you are honestly getting 429 with no abuse, then you can try this:

https://github.com/sietseringers/MACAddressTool

but honestly I dont believe you.

**Comment by Brojustuserandomgenerationusername** [comment]
Now i stopped having 429 errors out of nowhere. When i download some videos, however, i download 0byte sized files, this only applies to some videos, mostly music. Is this YT blocking video downloaders?

**Comment by Brojustuserandomgenerationusername** [comment]
> 
> 
> Read the pinned issue including the linked instructions, that you should have read before opening the issue.

BTDT.

**Comment by dirkf** [comment]
> > Read the pinned issue including the linked instructions, that you should have read before opening the issue.
> BTDT.

But apparently ignored the instructions.

If you have a problem URL, say 0 byte download, you could report it in a separate issue where it might be debugged.

**Comment by Brojustuserandomgenerationusername** [comment]
> 
> 
> > > Read the pinned issue including the linked instructions, that you should have read before opening the issue.
> > > BTDT.
> 
> But apparently ignored the instructions.
> 
> If you have a problem URL, say 0 byte download, you could report it in a separate issue where it might be debugged.

I didn't. The issue was reproduced every time even with hours in between with any video, and the issue where now files are downloaded as corrupted 0 byte files came after i made the ticket.

**Comment by Atlas48** [comment]
See also: [/blob/master/README.md#http-error-429-too-many-requests-or-402-payment-required](../blob/master/README.md#http-error-429-too-many-requests-or-402-payment-required).

**Comment by Brojustuserandomgenerationusername** [comment]
> 
> 
> See also: [/blob/master/README.md#http-error-429-too-many-requests-or-402-payment-required](../blob/master/README.md#http-error-429-too-many-requests-or-402-payment-required).

So which service should i ask to unblock? My ISP? The ytdl website?

**Comment by dirkf** [comment]
>... open a browser and solve a CAPTCHA the service suggests [to] you and after that pass cookies to youtube-dl. ...

As in your original post, it's the YouTube site that you need to satisfy, by playing the video on the site (solving any CAPTCHA) and then passing the cookies set by the site to yt-dl.

If you can't play the video in your browser (and the video is valid), you can try complaining to YT (good luck).

**Comment by Brojustuserandomgenerationusername** [comment]
> 
> 
> > ... open a browser and solve a CAPTCHA the service suggests [to] you and after that pass cookies to youtube-dl. ...
> 
> As in your original post, it's the YouTube site that you need to satisfy, by playing the video on the site (solving any CAPTCHA) and then passing the cookies set by the site to yt-dl.
> 
> If you can't play the video in your browser (and the video is valid), you can try complaining to YT (good luck).

The issue is i CAN play the video on the browser and no captcha shows up.

**Comment by dirkf** [comment]
And then you extract the cookies set by YT (by whatever means works for your browser) and pass those to the yt-dl command?

And set the UA to match the browser?

**Comment by Brojustuserandomgenerationusername** [comment]
> 
> 
> And then you extract the cookies set by YT (by whatever means works for your browser) and pass those to the yt-dl command?
> 
> And set the UA to match the browser?

How would i get going to doing that on firefox?
